test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -991,35 +1144,35 @@ def test_cgroup_offline_node(self):
         # Write each PID into the tasks file for the freezer cgroup
         task_file = os.path.join(fdir, 'tasks')
         for task in tasks[1:]:
-            fn = self.du.create_temp_file(body=task)
-            ret = self.du.run_copy(self.momA, src=fn,
+            self.tempfile = self.du.create_temp_file(
+                hostname=self.serverA, body=task)
+            ret = self.du.run_copy(self.momA, src=self.tempfile,
                                    dest=task_file, sudo=True,
                                    uid='root', gid='root',
                                    mode=0644)
-            os.remove(fn)
             if ret['rc'] != 0:
                 self.skipTest("pbs_cgroups_hook: need root privileges")
         # Freeze the cgroup
         freezer_file = os.path.join(fdir, 'freezer.state')
         state = 'FROZEN'
-        fn = self.du.create_temp_file(body=state)
-        ret = self.du.run_copy(self.momA, src=fn,
+        self.tempfile = self.du.create_temp_file(
+            hostname=self.serverA, body=state)
+        ret = self.du.run_copy(self.momA, src=self.tempfile,
                                dest=freezer_file, sudo=True,
                                uid='root', gid='root',
                                mode=0644)
-        os.remove(fn)
         if ret['rc'] != 0:
             self.skipTest("pbs_cgroups_hook: need root privileges")
-        self.server.expect(NODE, {'state': 'offline'},
+        self.server.expect(NODE, {'state': (MATCH_RE, 'offline')},
                            id=self.momA, interval=3)
         # Thaw the cgroup
         state = 'THAWED'
-        fn = self.du.create_temp_file(body=state)
-        ret = self.du.run_copy(self.momA, src=fn,
+        self.tempfile = self.du.create_temp_file(
```

Comment:
You lose the previous file when you overwrite self.tempfile, you need to remove it before this line.